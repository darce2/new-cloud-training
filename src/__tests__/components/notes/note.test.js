import React from 'react';
import Note from '../../../components/notes/note';

import { shallow } from 'enzyme';
import ContentEditable from 'react-contenteditable';

describe('App', () => {

  const mockHandleContentChange = jest.fn();
  const mockHandleDeleteNote = jest.fn();
  const mockHandleTitleChange = jest.fn();

  let props;

  beforeEach(() => {
    mockHandleContentChange.mockReset();
    mockHandleTitleChange.mockReset();
    mockHandleDeleteNote.mockReset();

    props = {
      title: "hi",
      content: "bye",
      index: 5,
      handleContentChange: mockHandleContentChange,
      handleDeleteNote: mockHandleDeleteNote,
      handleTitleChange: mockHandleTitleChange
    }
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<Note {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('handles delete', () => {
    const wrapper = shallow(<Note {...props} />);
    wrapper.find(".note__title").simulate("change", { target: { value: "foo" }});
    expect(mockHandleTitleChange).toHaveBeenCalled();
  });

  it('handles content', () => {
    const wrapper = shallow(<Note {...props} />);
    wrapper.find(ContentEditable).at(1).simulate("change", { target: { value: "foo" }});
    expect(mockHandleContentChange).toHaveBeenCalled();
  });

  it('handles delete', () => {
    const wrapper = shallow(<Note {...props} />);
    wrapper.find(".note__btn").simulate("click");
    expect(mockHandleDeleteNote).toHaveBeenCalled();
  });

});