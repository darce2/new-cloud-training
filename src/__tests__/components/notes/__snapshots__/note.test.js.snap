// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `ShallowWrapper {}`;
